src/XamlIl.Cecil/CecilMethod.cs

```diff
@@ -80,13 +89,30 @@ class CecilMethod : CecilMethodBase, IXamlIlMethodBuilder
             {
             }
 
+            private CecilMethod(CecilTypeSystem typeSystem, GenericInstanceMethod method)
+                : base(typeSystem, method)
+            {
+            }
+
             public bool Equals(IXamlIlMethod other) =>
                 // I hope this is enough...
                 other is CecilMethod cm
                 && DeclaringType.Equals(cm.DeclaringType)
                 && Reference.FullName == cm.Reference.FullName;
 
+            public IXamlIlMethod MakeGenericMethod(IReadOnlyList<IXamlIlType> typeArguments)
+            {
+                GenericInstanceMethod instantiation = new GenericInstanceMethod(Reference);
+                if (Reference == Definition)
+                {
+                    foreach (var type in typeArguments.Cast<ITypeReference>().Select(r => r.Reference))
+                    {
+                        instantiation.GenericArguments.Add(type);
```

Comment:
Please, check if this works with non-imported types. We might need to import them separately and rewrite the reference during IL generation [here](https://github.com/kekekeks/XamlIl/blob/bcb3a6583bff54f103180899bd43d699f62478a9/src/XamlIl.Cecil/CecilEmitter.cs#L126) like we do for types: https://github.com/kekekeks/XamlIl/blob/bcb3a6583bff54f103180899bd43d699f62478a9/src/XamlIl.Cecil/CecilEmitter.cs#L21

Reply:
I'm working on adding the importing.

Any particular reason that we manually recursively import (like importing the field type after we import the field)? It looks like Cecil does the recursive import for us as per their sources.

Reply:
Iterated on this and got it working correctly for generic methods referencing types outside of the current module.

Reply:
>Any particular reason

It complains about not imported types on module saving otherwise. Not sure if that's a bug in Cecil or Cecil working as intended.

Reply:
Yeah I saw that when working on the second commit.